MessageCenter3/Models/UserGroup/IUserGroupsRepository.cs

```diff
@@ -8,13 +8,14 @@ namespace MessageCenter3.Models
     public interface IUserGroupsRepository
     {
         List<UserGroup> GetAllUserGroups();
-        List<UserGroup> GetUserGroupsByUserId(int userId);
+        List<UserGroup> GetUserGroupsByUserId(int userId);        
+        List<UserGroup> GetAvailableUserGroups(int userId);
         UserGroup GetUserGroupById(int groupId);
         List<UserGroup> GetUserGroupsByName(string searchName);
         UserGroup AddUserGroup(UserGroupInputModel newGroup);
         UserGroup RenameUserGroup(UserGroup group);
-        bool AddUserToGroup(int userId, int groupId);
-        void ExcludeUserFromGroup(int userId, int groupId);
+        void AddUserToGroup(JoinGroupRequestModel request);
+        void ExcludeUserFromGroup(JoinGroupRequestModel request);
         bool DeleteUserGroup(int groupId);
```

Comment:
I believe it should be separated GroupRepository

Reply:
It actually is a (User)GroupRepository